### PR TITLE
prometheus-exporter: Add retry mechanism to reach pushgateway

### PR DESCRIPTION
**Pull request by surajssd**
Adds a retry logic to reach to pushgateway.

### CODE REVIEW

**Review by t-lo — APPROVED** [review]
LGTM, thanks!

**Comment by surajssd** [comment]
I will hold on to this PR until I fix the error properly and test and verify it.

File: docker/prometheus-export-wrapper [hunk 1]
```diff
@@ -490,3 +502,15 @@ awk '
...
+until [ "$n" -ge 10 ]
+do
+    if pgrep linkerd2-proxy; then
+        kill $(pgrep linkerd2-proxy)
+        sleep 3
+    fi
+    n=$((n+1))
+    echo "killing linkerd2-proxy"
+    sleep 1
+done
```
Comment (t-lo) [inline docker/prometheus-export-wrapper:523]: How about

```$shell
grace_max=10
grace="$grace_max"
while true; do
    pkill linkerd2-proxy
    if [ $? -eq 0 ]; then
        sleep 1
        grace="$grace_max"
        continue
    fi

    if [ $grace -gt 0 ]; then
        sleep 1
        echo "Linkerd-proxy shutdown grace period: $grace"
        grace=$(($grace-1))
        continue
    fi
    break
done

# Rule #2: Double Tap
pkill -9 linkerd2-proxy

# exit successfully
true
```

Comment (surajssd) [reply docker/prometheus-export-wrapper:523]: done.

**Review by t-lo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**